docs/samples/autoscaling/README.md

```diff
@@ -0,0 +1,306 @@
+
+### Create the KFService
+Apply the tensorflow example CR
+```
+kubectl apply -f autoscale.yaml
+```
+
+Expected Output
+```
+$ kfservice.serving.kubeflow.org/flowers-sample configured
+```
+
+### Load KFService with concurrent requests
+Send traffic in 30 seconds spurts maintaining 5 in-flight requests.
+```
+MODEL_NAME=flowers-sample
+INPUT_PATH=../tensorflow/input.json
+CLUSTER_IP=$(kubectl -n istio-system get service istio-ingressgateway -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
+HOST=$(kubectl get kfservice $MODEL_NAME -o jsonpath='{.status.url}')
+hey -z 30s -c 5 -m POST -host ${HOST} -D $INPUT_PATH http://$CLUSTER_IP/v1/models/$MODEL_NAME:predict
+```
+Expected Output
+```shell
+Summary:
+  Total:	30.0193 secs
+  Slowest:	10.1458 secs
+  Fastest:	0.0127 secs
+  Average:	0.0364 secs
+  Requests/sec:	137.4449
+  
+  Total data:	1019122 bytes
+  Size/request:	247 bytes
+
+Response time histogram:
+  0.013 [1]	|
+  1.026 [4120]	|■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■
+  2.039 [0]	|
+  3.053 [0]	|
+  4.066 [0]	|
+  5.079 [0]	|
+  6.093 [0]	|
+  7.106 [0]	|
+  8.119 [0]	|
+  9.133 [0]	|
+  10.146 [5]	|
+
+
+Latency distribution:
+  10% in 0.0178 secs
+  25% in 0.0188 secs
+  50% in 0.0199 secs
+  75% in 0.0210 secs
+  90% in 0.0231 secs
+  95% in 0.0328 secs
+  99% in 0.1501 secs
+
+Details (average, fastest, slowest):
+  DNS+dialup:	0.0002 secs, 0.0127 secs, 10.1458 secs
+  DNS-lookup:	0.0002 secs, 0.0000 secs, 0.1502 secs
+  req write:	0.0000 secs, 0.0000 secs, 0.0020 secs
+  resp wait:	0.0360 secs, 0.0125 secs, 9.9791 secs
+  resp read:	0.0001 secs, 0.0000 secs, 0.0021 secs
+
+Status code distribution:
+  [200]	4126 responses
+```
+Check the number of running pods now, `KFServing` uses `Knative Serving` autoscaler which is based on the
+average number of in-flight requests per pod(concurrency). `KFServing` by default sets target concurrency to be 1 and we
+load the service with 5 concurrent requests, so the autoscaler tries scaling up to 5 pods. Notice that out of all the requests there are 
+5 requests on the histogram that take around 10s, that's the cold start time cost to initially spawn the pods and download model to be ready
+to serve. The cold start may take longer(to pull the serving image) if the image is not cached on the node that the pod is scheduled on.
+```
+$ kubectl get pods
+NAME                                                       READY   STATUS            RESTARTS   AGE
+flowers-sample-default-7kqt6-deployment-75d577dcdb-sr5wd         3/3     Running       0          42s
+flowers-sample-default-7kqt6-deployment-75d577dcdb-swnk5         3/3     Running       0          62s
+flowers-sample-default-7kqt6-deployment-75d577dcdb-t2njf         3/3     Running       0          62s
+flowers-sample-default-7kqt6-deployment-75d577dcdb-vdlp9         3/3     Running       0          64s
+flowers-sample-default-7kqt6-deployment-75d577dcdb-vm58d         3/3     Running       0          42s
+```
```

Comment:
a mini comment, the pod should have 2 containers instead of 3

Reply:
oh I have istio sidecar injected.